src/controllers/permissionController.js

```diff
@@ -0,0 +1,118 @@
+import { Role, Permission } from '../sequelize/models';
+
+/**
+ * @description Users role and thier permission
+ */
+class permissionController {
+  /**
+   *
+   * @param {object} req
+   * @param {object} res
+   * @returns {object} return object containing a new created permission
+   */
+  static async createPermision(req, res) {
+    const { role } = req.params;
+    const newPermission = req.body.permission;
+
+    const { id } = await Role.findOne({ where: { role } });
+    const permission = await Permission.findOne({ where: { roleId: id } });
+
+    if (permission && permission.roleId === id && permission.permission === newPermission) {
+      return res.status(409).json({
+        error: 'This permission is already defined'
+      });
+    }
+    const payload = {
+      roleId: id,
+      permission: newPermission,
+    };
+
+    const createPermission = await Permission.create(payload);
+    return res.status(201).json({
+      permission: {
+        id: createPermission.id,
+        role,
+        permission: createPermission.permission,
+        createdAt: createPermission.createdAt,
+        updatedAt: createPermission.updatedAt
+      }
+    });
+  }
+
+  /**
+   *
+   * @param {object} req
+   * @param {object} res
+   * @returns {object} return object containing a list of permission to the different useres
+   */
+  static async getRolePermissions(req, res) {
+    const { role } = req.params;
+    const { id } = await Role.findOne({ where: { role } });
+
+    const permissions = await Permission.findAll({ where: { roleId: id } });
+
+    if (permissions.length === 0) {
```

Comment:
I think you can use something like `!permissions.length` here.

Reply:
Why? findAll() return array. For example array[], its length is zero. So for `permissions.length === 0` I just wanted to test if length of that array equals to zero.

Reply:
Yeah, I know it returns an array but that condition simplifies and covers it also.

Reply:
Yes I see, but all serve the same purpose. Is that way the best practice?

Reply:
Done